data_source_search_track.go

```diff
@@ -156,19 +161,28 @@ func dataSourceSearchTrackRead(d *schema.ResourceData, m interface{}) error {
 	var ids []string
 	for _, track := range results.Tracks.Tracks {
 		var artists []interface{}
-		for _, artist := range track.Artists {
-			artists = append(artists, artist.ID.String())
-		}
-		tracks = append(tracks, map[string]interface{}{
+		trackData := map[string]interface{}{
 			"id":      track.ID.String(),
 			"name":    track.Name,
 			"artists": artists,
 			"album":   track.Album.ID.String(),
-		})
+		}
+		for _, artist := range track.Artists {
+			artists = append(artists, artist.ID.String())
+		}
```

Comment:
Thanks for the PR! I think there's a small bug here. You set the artists in the map, then you update the artists later. Sometimes this works sometimes it doesn't. I'd swap around the trackData/artists loop